Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/SettingsFile.java

```diff
@@ -398,6 +403,31 @@ private static SettingSection sectionFromLine(String line)
 		return new SettingSection(sectionName);
 	}
 
+	private static void addGcPadSettingsIfTheyDontExist(HashMap<String, SettingSection> sections)
+	{
+		SettingSection coreSection;
+		if (sections.isEmpty() || sections.get(SettingsFile.SECTION_CORE) == null)
```

Comment:
`sections` is a `SettingsSectionMap`, which automatically `put`s a section if you try to `get` a section that doesn't exist. So you should be able to skip these checks and just call `sections.get(SettingsFile.SECTION_CORE)`.